### PR TITLE
test: OTEL mTLS E2E Test for TracePipeline and MetricPipeline

### PR DESCRIPTION
**Pull request by dennis-ge**
## Description

Changes proposed in this pull request:

- implement an E2E test for the OTEL mTLS feature introduced in https://github.com/kyma-project/telemetry-manager/pull/347
- add the test case to the TracePipeline as well as to the MetricPipeline
- refactor the e2e traces and metrics tests a bit by extracting common functionality in functions and introducing the options patterns for the mock backend

## Related Issues and Documents

Closes:

Related issues: https://github.com/kyma-project/kyma/issues/17995

## Traceability

- [ ] The PR is linked to a GitHub Issue.
- [ ] New features have a milestone label set.
- [ ] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub Issue has a respective `area` label.
- [x] The follow-up issues (if any) are linked in the `Related Issues` section.

## Testability

The feature is unit-tested:

- [ ] Yes.
- [x] No, because unit tests are not needed.
- [ ] No, because of ...

The feature is e2e-tested:

- [x] Yes.
- [ ] No, because e2e-tests are not needed.
- [ ] No, because of ...

Tests conducted for the PR:

## Codebase

- [ ] My code follows the [Effective Go](https://go.dev/doc/effective_go) style guidelines.
- [ ] The code was planned and designed following the defined architecture and the separation of concerns.
- [x] The code has sufficient comments, particularly for all hard-to-understand areas.
- [ ] This PR adds value and shows no feature creep.
- [x] I have augmented the test suite that proves my fix is effective or that my feature works.
- [ ] Adjusted the documentation if the change is user-facing.

### CODE REVIEW

**Review by hisarbalik — DISMISSED** [review]

File: test/testkit/mocks/backend.go
Comment (hisarbalik) [inline test/testkit/mocks/backend.go:1]: Could you please rebase and solve the merge conflicts

File: test/testkit/mocks/backend.go [hunk 1]
```diff
@@ -40,6 +40,10 @@ func (b *Backend) HTTPBackendConfigMap(name string) *LogBackendConfigMap {
 	return NewLogBackendConfigMap(name, b.namespace, b.exportedFilePath)
 }
 
+func (b *Backend) TLSBackendConfigMap(name, certPem, keyPem, caPem string) *TLSBackendConfigMap {
+	return NewTLSBackendConfigMap(name, b.namespace, b.exportedFilePath, certPem, keyPem, caPem, b.signalType)
+}
```
Comment (skhalash) [inline test/testkit/mocks/backend.go:45]: Maybe let's instead pass the TLS option into the `Backend` constructor function and use the old `ConfigMap` function?

**Review by skhalash — APPROVED** [review]

**Pull request merged**